src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1151,27 +1151,91 @@ MonodroidRuntime::monodroid_dlopen (const char *name, int flags, char **err, [[m
 {
 	unsigned int dl_flags = monodroidRuntime.convert_dl_flags (flags);
 	bool libmonodroid_fallback = false;
-
+	bool name_is_full_path = false;
+	bool name_needs_free = false;
 	/* name is nullptr when we're P/Invoking __Internal, so remap to libxa-internal-api */
-	if (name == nullptr) {
+	if (name == nullptr || strstr (name, "xa-internal-api") != nullptr) {
```

Comment:
Should this instead be `strstr (name, MONODROID_PATH_SEPARATOR "xa-internal-api") != nullptr`, ensuring that we get e.g. `/xa-internal-api`?

Reply:
I'd rather not do that. We can't be sure we'll always be getting a path here. Right now we do get **a** path (which points to a location where there's no libraries we ship) but it may change and I'd rather be flexible.